source/filter.js

```diff
@@ -0,0 +1,37 @@
+"use strict";
+
+const map = {
+  "&": "&amp;",
+  "<": "&lt;",
+  ">": "&gt;",
+  '"': "&quot;",
+  "'": "&#39;",
+};
+
+
+/**
+ * Экранирует символы в html, оставляя только разрешенные теги
+ * @param {string} text - фильтруемый текст
+ * @param {Array<string>} tags - список разрешенных тегов
+ * @returns {string} отфильтрованный текст
+ */
+const filter = (text, tags = []) => {
+  if (!(typeof text === "string")) {
+    throw new Error("expected String");
+  }
+  if (!Array.isArray(tags) || tags.some((el) => typeof el !== "string")) {
+    throw new Error("expected Array of strings");
+  }
+  const resArr = text.split(/([&"<>'])/g);
+  for (let i = 1; i < resArr.length; i += 2) {
+    if (
+      resArr[i] == "<" && resArr[i + 2] == ">" &&
```

Comment:
```suggestion
      resArr[i] === "<" && resArr[i + 2] === ">" &&
```
потому что теперь это точно строки и не будет заставлять того кто это будет читать задаваться вопросом "а почему не 3" -> "тут не в точности это строка"

Reply:
По сути приемлемо писать `==` только при проверке сразу и на null и на undefined и даже когда был объект лучше/понятнее было бы достать конкретное значение из объекта - потому что там иначе происходила конвертация массива в строку, а это не имеет особо смысла

Но это просто - справочная информация =)